src/main/java/cook/Cook.java

```diff
@@ -0,0 +1,154 @@
+package cook;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+
+import exceptions.InvalidCommandException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.ToDo;
+
+/**
+ * Cook class to store main logic and program.
+ */
+public class Cook {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructor for Cook class.
+     */
+    public Cook(File file) {
+        this.storage = new Storage(file);
+        this.tasks = new TaskList();
+        this.ui = new Ui();
+
+        HashMap<String, Integer> validCommandArgs = new HashMap<>();
+
+        validCommandArgs.put("bye", 1);
+        validCommandArgs.put("list", 1);
+        validCommandArgs.put("mark", 2);
+        validCommandArgs.put("unmark", 2);
+        validCommandArgs.put("delete", 2);
+        validCommandArgs.put("find", 2);
+        validCommandArgs.put("todo", 2);
+        validCommandArgs.put("deadline", 4);
+        validCommandArgs.put("event", 6);
+
+        this.parser = new Parser(validCommandArgs);
+    }
+
+    /**
+     * Runs main logic.
+     */
+    public void run() {
+        this.ui.welcome();
+
+        while (true) {
+            String input = this.ui.getInput();
+            HashMap<String, String> argumentsHashMap;
+
+            try {
+                argumentsHashMap = this.parser.readInput(input);
+            } catch (InvalidCommandException e) {
+                this.ui.say(e.getMessage());
+                continue;
+            }
+
+            String command = argumentsHashMap.get("command");
+            if (command.equals("bye")) {
+                return;
+            } else if (command.equals("list")) {
+                this.ui.say(this.tasks.toString());
+            } else if (command.contains("mark") || command.equals("delete")) {
+                int taskNo;
```

Comment:
While in general people will understand "no" as "number", perhaps you could use the longer form of the word to avoid possible confusion and improve code quality?

Reply:
🫡